submission/score_model.py

```diff
@@ -0,0 +1,145 @@
+import datetime
+import json
+import logging
+import os
+import subprocess
+import sys
+import zipfile
+import git
+from pathlib import Path
+
+from importlib import import_module
+
+from brainscore.utils import LazyLoad
+
+from submission import score_model
+from submission.ml_pool import MLBrainPool, ModelLayers
+
+logger = logging.getLogger(__name__)
+
+all_benchmarks_list = [
+    'movshon.FreemanZiemba2013.V1-pls',
+    'movshon.FreemanZiemba2013.V2-pls',
+    'dicarlo.Majaj2015.V4-pls', 'dicarlo.Majaj2015.IT-pls',
+    'dicarlo.Rajalingham2018-i2n',
+    # 'dicarlo.Kar2019-ost',
+    'fei-fei.Deng2009-top1'
+]
+
+
+def score_models(config_file, work_dir, db_connection_config, jenkins_id, models=None,
+                 benchmarks=None):
+    config_file = config_file if os.path.isfile(config_file) else os.path.realpath(config_file)
+    db_conn = connect_db(db_connection_config)
+    with open(config_file) as file:
+        configs = json.load(file)
+    logger.info(f'Run with following configurations:{str(configs)}')
+    if configs['type'] == 'zip':
+        config_path = Path(config_file).parent
+        logger.info('Start executing models in repo %s' % (configs['zip_filename']))
+        repo = extract_zip_file(configs, config_path, work_dir)
+    else:
+        logger.info('Start executing models in repo %s' % (configs['git_url']))
+        repo = clone_repo(configs, work_dir)
+    package = 'models.brain_models' if configs['model_type'] == 'BrainModel' else 'models.base_models'
+    module = install_project(repo, package)
+    test_benchmarks = all_benchmarks_list if benchmarks is None or len(benchmarks) == 0 else benchmarks
+    ml_brain_pool = {}
+    if configs['model_type'] == 'BaseModel':
+        test_models = module.get_model_list() if models is None or len(models) == 0 else models
+        logger.info(f"Start working with base models")
+        layers = {}
+        base_model_pool = {}
+        for model in test_models:
+            function = lambda: module.get_model(model)
+            base_model_pool[model] = LazyLoad(function)
+            if module.get_layers is not None:
```

Comment:
Yes this will then just fail when we try to run the model...which gives give an error message in the try catch block. I don't really want to break the whole process only because the user screwed it up.